core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -131,12 +135,23 @@ public InternalEngine(EngineConfig engineConfig) throws EngineException {
         boolean success = false;
         try {
             this.lastDeleteVersionPruneTimeMSec = engineConfig.getThreadPool().estimatedTimeInMillis();
-            seqNoService = new SequenceNumbersService(shardId, engineConfig.getIndexSettings());
+
             mergeScheduler = scheduler = new EngineMergeScheduler(engineConfig.getShardId(), engineConfig.getIndexSettings());
             throttle = new IndexThrottle();
             this.searcherFactory = new SearchFactory(logger, isClosed, engineConfig);
             try {
                 writer = createWriter(openMode == EngineConfig.OpenMode.CREATE_INDEX_AND_TRANSLOG);
+                final long localCheckpoint = loadLocalCheckpointFromCommit(writer);
```

Comment:
Nit: maybe load a `SeqNoStats` with one method? this is probably how it's going to be?